src/Esprima/Token.cs

```diff
@@ -116,12 +119,20 @@ private sealed record TemplateHolder(string? Cooked, string RawTemplate, char No
     public readonly int LineNumber;
     public readonly int LineStart;
 
-    public object? Value => Type switch
+    public object? Value
     {
-        TokenType.Template => ((TemplateHolder) _value!).Cooked,
-        TokenType.RegularExpression => ((RegexHolder) _value!).Value,
-        _ => _value
-    };
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get
+        {
+            // NOTE: This condition must not be inverted, otherwise the runtime (.NET 6) fail to inline the accessor correctly.
+            return Type is not (TokenType.RegularExpression or TokenType.Template or TokenType.Extension)
+                ? _value
+                : GetValueFromHolder(in this);
+
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            static object? GetValueFromHolder(in Token token) => ((ValueHolder) token._value!).Value;
+        }
```

Comment:
Finally I've managed to find a combination which the runtime can inline without penalty.

If you change *anything* in this code (even if just invert the condition), inlining goes south and the acceptance test starts to fail. Mind-boggling...

However, this looks to work reliably and according to the benchmarks, we even have a little bit of improvement in execution time:

## Esprima.Benchmark.FileParsingBenchmark

| **Diff**|Method|FileName|Mean|Error|Gen 0|Gen 1|Gen 2|Allocated|
|------- |-------|-------|-------:|-------|-------:|-------:|-------:|-------:|
| Old |ParseProgram|angular-1.2.5|17.628 ms|2.5634 ms|718.7500|343.7500|-|4,315 KB|
| **New** |	|	| **17.475 ms (-1%)** | **1.1170 ms** | **718.7500 (0%)** | **312.5000 (-9%)** | **-** | **4,315 KB (0%)** |
| Old |ParseProgram|backbone-1.1.0|2.408 ms|0.2234 ms|136.7188|39.0625|-|676 KB|
| **New** |	|	| **2.338 ms (-3%)** | **0.2539 ms** | **140.6250 (+3%)** | **35.1563 (-10%)** | **-** | **676 KB (0%)** |
| Old |ParseProgram|jquery-1.9.1|14.161 ms|0.6347 ms|625.0000|296.8750|-|3,764 KB|
| **New** |	|	| **13.805 ms (-3%)** | **1.0131 ms** | **609.3750 (-3%)** | **296.8750 (0%)** | **-** | **3,764 KB (0%)** |
| Old |ParseProgram|jquery.mobile-1.4.2|22.922 ms|0.3971 ms|1031.2500|500.0000|31.2500|5,798 KB|
| **New** |	|	| **22.039 ms (-4%)** | **2.4387 ms** | **1031.2500 (0%)** | **500.0000 (0%)** | **31.2500 (0%)** | **5,798 KB (0%)** |
| Old |ParseProgram|mootools-1.4.5|11.000 ms|0.1603 ms|500.0000|250.0000|-|3,091 KB|
| **New** |	|	| **10.776 ms (-2%)** | **0.4187 ms** | **500.0000 (0%)** | **250.0000 (0%)** | **-** | **3,091 KB (0%)** |
| Old |ParseProgram|underscore-1.5.2|2.022 ms|0.0332 ms|117.1875|42.9688|-|571 KB|
| **New** |	|	| **1.930 ms (-5%)** | **0.0355 ms** | **113.2813 (-3%)** | **42.9688 (0%)** | **-** | **571 KB (0%)** |
| Old |ParseProgram|yui-3.12.0|10.352 ms|1.6380 ms|468.7500|234.3750|-|2,856 KB|
| **New** |	|	| **9.828 ms (-5%)** | **1.0396 ms** | **468.7500 (0%)** | **234.3750 (0%)** | **-** | **2,856 KB (0%)** |